graph/vertex/impl/ThingVertexImpl.java

```diff
@@ -329,4 +330,70 @@ public void setModified() {
             }
         }
     }
+
+    /**
+     * A Target vertex can emulate a Read or Write vertex for the purposes of seeking to a particular
+     * vertex that may or may not exist on disk or in transaction-local write-vertex buffers. However
+     * it should not be usable for anything else.
+     */
+    public static class Target extends ThingVertexImpl implements ThingVertex.Write {
```

Comment:
we introduce a `Target` vertex used for finding vertices that match some value that may not exist. It must be a `Write` vertex because we want to be able to search through both persisted or in-memory vertices.